packages/@apphosting/common/src/index.ts

```diff
@@ -31,26 +31,31 @@ interface ServerConfig {
   maxInstances?: number;
 }
 
-// Additonal fields needed for adapter identification
+// Additonal fields needed for identifying the framework and adapter being used
 interface Metadata {
-  // Name of the adapter (this should be the npm package name)
-  adapterNpmPackageName: string;
-  // Version of the adapter
+  // Name of the adapter (this should be the official package name) e.g. "@apphosting/adapter-nextjs"
+  adapterPackageName: string;
+  // Version of the adapter, e.g. "18.0.1"
   adapterVersion?: string;
```

Comment:
why optional?

Reply:
Originally this field was added so authors had the option to tell us what version if they wanted to (technically this is something we can self discover on the buildpacks side too) but I think it's pretty straightforward for adapter authors to provide this to us as well (done here as an example https://github.com/FirebaseExtended/firebase-framework-tools/pull/248/files#diff-c4645e334899a380f959aecbba2ce11a7ffcd895b12daed80af12d265eb9d437R120-R125) 

I've gone ahead and made it required so we don't need to do it in the buildpacks anymore